### DISCUSSION TITLE
[ZK Stack] - <Reth Node Goes Offline Due to 'Could Not Find Block or TX Header>

### DISCUSSION THREAD

**Discussion by Szczepoo13** [zkSync-Community-Hub/zksync-developers#694] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.83.0-nightly

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

While running a local reth node inside of a docker container - after some time of running (with command **zk_inception server**), the node shuts down due to the error: “**Could not find block or tx number on a range request segment = Headers number=392365** ”. This error is causing the node to go offline/ into restart mode.
![nodedown](https://github.com/user-attachments/assets/c7630e6b-6143-4b97-b742-2ead22d0358f)

### Expected Behavior

Node runs normally, do not enter the restart state.

### Current Behavior

While running command **zk_inception server** after some time the node shuts down and goes into offline/restart mode - not allowing a server to be run successfully.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
It seems like the issue you're encountering with the Reth node going offline due to a missing block or transaction header is related to a problem with the node's ability to maintain a stable connection to the blockchain data it requires. This could be due to several factors, such as network issues, configuration errors, or bugs in the software.

Here are a few steps you can take to troubleshoot and potentially resolve the issue:

1. **Check Network Connectivity**: Ensure that your Docker container has proper network settings that allow it to communicate with the blockchain network effectively.
2. **Review Docker Container Logs**: Look at the logs of your Docker container to see if there are any additional errors or warnings that might give more context about the problem.
3. **Update and Restart**: Ensure that your ZKsync and Reth node software are up to date. Sometimes, bugs are fixed in newer versions of the software.
4. **Configuration Review**: Double-check the configuration settings of your node. Make sure that all configurations are correct and compatible with the versions of the software you are using.

If the problem persists, consider reaching out for more specialized help or consulting the documentation for more advanced troubleshooting techniques.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by bxpana** [comment]
Has this been a frequent occurrence or just happened for the first time?

**Reply by Szczepoo13** [reply]
Hey, unfortunately, it is a frequent occurrence. It has happened to me on many different instances and at various times.
The node ran, but after some time, I checked on it again, and it was already in the restarting phase.

**Reply by bxpana** [reply]
Ok let me see if I can find out anything else.

**Reply by Szczepoo13** [reply]
Hey, I found something like the **'zk_supervisor'** tool in the toolbox to manipulate databases (however I cant find installations) - for me it seems like the issue might be that the Reth node failed to sync or diverged too much. Maybe, resetting the node to a fresh state could help get it running again?

**Comment by SuccinctPaul** [comment, marked as answer]
Hi, meet the same error and fixed the error by upgrading the version of reth  image(>=v1.0.0).

And the zksync-era has fixed the error in the latest code(https://github.com/SuccinctPaul/zksync-era/commit/fe0867732f65459d366ffa029e87b17482574117#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3)

* More reference.
https://github.com/paradigmxyz/reth/issues/7516#issuecomment-2225124805
https://github.com/paradigmxyz/reth/issues/7618#issuecomment-2106229317
